polaris-core/src/main/java/io/polaris/core/persistence/PolarisMetaStoreManager.java

```diff
@@ -176,18 +176,28 @@ public boolean alreadyExists() {
   }
 
   /**
-   * Bootstrap the Polaris service, will remove ALL existing persisted entities, then will create
-   * the root catalog, root principal and associated service admin role.
+   * Bootstrap the Polaris service, creating the root catalog, root principal, and associated
+   * service admin role. Will fail if the service has already been bootstrapped.
+   *
+   * @param callCtx call context
+   * @return always success or unexpected error
+   */
+  @NotNull
+  BaseResult bootstrapPolarisService(@NotNull PolarisCallContext callCtx);
```

Comment:
Currently, the behavior is to throw an `IllegalArgumentException` if the realm cannot be bootstrapped. It seems for consistency with the other APIs, it should return the `BaseResult` with `success=false` rather than throwing the exception. Let the `bootstrap` command deal with the failure state